cmd/zc_traverser_local.go

```diff
@@ -197,11 +197,23 @@ func WalkWithSymlinks(appCtx context.Context, fullPath string, walkFunc filepath
 				computedRelativePath = ""
 			}
 
+			if (fileInfo.Mode()&os.ModeSocket != 0) || (fileInfo.Mode()&os.ModeNamedPipe != 0) || (fileInfo.Mode()&os.ModeIrregular != 0) {
+				WarnStdoutAndScanningLog(fmt.Sprintf("Unsupported file type %s: %v", filePath, fileInfo.Mode()))
+				return nil
+			}
+
 			if fileInfo.Mode()&os.ModeSymlink != 0 {
 				if !followSymlinks {
 					return nil // skip it
 				}
-				result, err := UnfurlSymlinks(filePath)
+
+				/*
+				 * There is one case where symlink can point to outside of sharepoint. In that case
+				 * we need to throw error. Its very likely same file or folder present on the agent side.
+				 *
+				 * TODO: Need to handle this case.
+				 */
+				result, err := filepath.EvalSymlinks(filePath)
```

Comment:
This should be fine for Linux but for Windows hosts we might need the specific handling done by UnfurlSymlinks (common.ToExtendedPath).
I think it'll be better to update UnfurlSymlinks() to simply call filepath.EvalSymlinks() if not Windows, else let the current handling continue, to avoid making any behavior change.

Also, I hope this will be fine for SMB mount on Linux. Will need to think more.

@zezha-msft, thoughts?